RecoTracker/FinalTrackSelectors/plugins/TrackListMerger.cc

```diff
@@ -446,12 +444,20 @@ void TrackListMerger::produce(edm::Event& e, const edm::EventSetup& es) {
     }    //end more than 0 track
   }      // loop over trackcolls
 
+  if (ngood == 0) {
+    // output empty collections and early return
+    this->returnEmptyCollections(e);
+    return;
+  }
+
+  statCount.begin(rSize);
   statCount.pre(ngood);
 
   //cache the id and rechits of valid hits
   typedef std::pair<unsigned int, const TrackingRecHit*> IHit;
   std::vector<std::vector<IHit>> rh1(ngood);  // "not an array" of vectors!
   //const TrackingRecHit*  fh1[ngood];  // first hit...
+
```

Comment:
Sorry to be a pain, but you can please revert the added newline just to clean up the diff?  Thanks :-)

Reply:
you're welcome to do it later :)

Reply:
It would be better if you take care of it now, after which I'll sign.

Reply:
as far as I am concerned this can stay open forever, not planning to push more unless there are physics relevant comments.